apis/v1alpha2/validation/httproute.go

```diff
@@ -166,3 +146,32 @@ func validateHTTPPathMatch(path *gatewayv1a2.HTTPPathMatch, fldPath *field.Path)
 	}
 	return allErrs
 }
+
+// validateHTTPRouteFilterTypeMatchesValue validates that each filter type matches
+//the filter value.
+func validateHTTPRouteFilterTypeMatchesValue(filters []gatewayv1a2.HTTPRouteFilter, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	for _, filter := range filters {
+		switch filter.Type {
```

Comment:
That current error is coming from this switch. 
```
 apis/v1alpha2/validation/httproute.go:155:3: missing cases in switch of type v1alpha2.HTTPRouteFilterType: HTTPRouteFilterURLRewrite (exhaustive)
		switch filter.Type { 
```
But I'm not getting that error when I run this verification locally. Also, that switch statement has a `default` clause so it doesn't make sense to get this error. Any ideas what could be the reason?

Reply:
It looks like this is complaining about a missing case for the `URLRewrite` filter type. That may have been added while you were working on this PR, so would've been easy to miss.

Reply:
Great catch by the linter. I guess the compiler evaluates the `default:` as correct but the linter is a bit more clever. 👍🏾